source/mainnet/smart-contracts/general/contract-instances.rst

```diff
@@ -168,3 +168,88 @@ If an operations fails, it returns an error, which the instance can choose to
 handle, and the state and balance of the instance remain unchanged.
 The account which sent the initiating transaction pays for the execution of the
 entire receive function, including the cost of failed operations.
+
+.. _contract-instance-upgradeability:
+
+Upgradeability
+==============
+
+Smart contract instances can choose to upgrade its module to a new smart contract
+module using the **upgrade** host function.
+The host function takes a reference to a deployed smart contract module to use for
+the upgraded instance and can only be called from a receive function.
+The function returns back whether the upgrade is successful or failed, allowing the instance
+to decide the next step. If the upgrade is succeful any new invocations of the
+upgraded instance uses the smart contract code in the new module.
+
+.. warning::
+
+   Make sure to proper authorized any endpoint triggering a smart contract upgrade as
+   it allows for changing the behavior of the smart contract.
+
+.. graphviz::
+   :align: center
+   :caption: Example of a smart contract instance 'Car' upgrading to a new module.
+
+   digraph G {
+       rankdir="BT"
+
+       subgraph cluster_0 {
+           label = "My module v1";
+           labelloc=b;
+           node [fillcolor=white, shape=note]
+           escrow;
+       }
+
+       subgraph cluster_2 {
+           label = "My module v2";
+           labelloc=b;
+           node [fillcolor=white, shape=note]
+           escrow2;
+       }
+
+       subgraph cluster_1 {
+           label = "Instances";
+           style=dotted;
+           node [shape=box, style=rounded]
+           House;
+           Car;
+       }
+
+       escrow[label="Escrow"]
+       escrow2[label="Escrow"]
+
+       House:n -> escrow;
+       Car:n -> escrow [style=dashed];
```

Comment:
I think `dashed` is a bit hard to distinguish from the solid line. Consider using a different style.

Reply:
Unfortunately, there are not that many options, but I have tried some other things to make it stand out a bit more.
Let me know what you think

Reply:
Your changes made it much better. Great!